internal/mode/static/nginx/runtime/manager.go

```diff
@@ -41,19 +41,29 @@ type Manager interface {
 	Reload(ctx context.Context, configVersion int) error
 }
 
+// ManagerCollector is an interface for the metrics of the NGINX runtime manager.
+type ManagerCollector interface {
+	IncNginxReloadCount()
+	IncNginxReloadErrors()
+	UpdateLastReloadTime(ms time.Duration)
+}
+
 // ManagerImpl implements Manager.
 type ManagerImpl struct {
-	verifyClient *verifyClient
+	verifyClient     *verifyClient
+	metricsCollector ManagerCollector
 }
 
 // NewManagerImpl creates a new ManagerImpl.
-func NewManagerImpl() *ManagerImpl {
+func NewManagerImpl(mgrCollector ManagerCollector) *ManagerImpl {
 	return &ManagerImpl{
-		verifyClient: newVerifyClient(nginxReloadTimeout),
+		verifyClient:     newVerifyClient(nginxReloadTimeout),
+		metricsCollector: mgrCollector,
 	}
 }
 
 func (m *ManagerImpl) Reload(ctx context.Context, configVersion int) error {
```

Comment:
I wonder if now we need it makes sense to test Reload? including its new side effects - increments of metrics?

cc @sjberman  @kate-osborn

Reply:
Yes, probably. How feasible is it, though? Can we test the metrics?

Reply:
Do you think it would be ok to punt on these tests for now? They'll likely take a bit of time, and it would be great to get this PR in before the rename. No worries if we don't though, but I think I'll need to recreate this PR from a new fork if it goes in after the rename so we might lose some context of the review comments 🤔

Reply:
shall we have an issue for it?